Engine/Main/GameObject.cpp

```diff
@@ -111,6 +112,32 @@ void GameObject::Delete(std::vector<GameObject*>& children_to_remove)
 		prefab_reference->RemoveInstance(this);
 	}
 }
+void GameObject::Duplicate(const GameObject & gameobject_to_copy)
+{
+	if (!is_prefab_parent && gameobject_to_copy.transform.modified_by_user)
+	{
+		transform.SetTranslation(gameobject_to_copy.transform.GetTranslation());
+		transform.SetRotation(gameobject_to_copy.transform.GetRotationRadiants());
+		//gameobject_to_copy.transform.modified_by_user = false;
+	}
+	transform.SetScale(gameobject_to_copy.transform.GetScale());
+	CopyComponents(gameobject_to_copy);
+	this->name = gameobject_to_copy.name;
+	this->active = gameobject_to_copy.active;
+	this->SetStatic(gameobject_to_copy.is_static);
+	this->hierarchy_depth = gameobject_to_copy.hierarchy_depth;
+	this->hierarchy_branch = gameobject_to_copy.hierarchy_branch;
+	this->original_UUID = gameobject_to_copy.original_UUID;
```

Comment:
I think when we duplicate gameobjects that are child prefabs you will need to maybe set the original_UUID  to 0 and the prefab_reference to nullptr. Because when duplicating a child prefab you will be adding a new child to the prefab that is not actually part of the original prefab.

Reply:
Okay I will check it! :D thanks!!